compiler/src/dmd/toobj.d

```diff
@@ -771,7 +771,21 @@ void toObjFile(Dsymbol ds, bool multiobj)
                 StringExp se = cast(StringExp)e;
                 char *name = cast(char *)mem.xmalloc(se.numberOfCodeUnits() + 1);
                 se.writeTo(name, true);
-
+                
+                Dsymbol symbol = getDsymbol(e);
+                if (name[0] == '.') {
+                    auto nameLen = strlen(name);
+                    auto filenameLen = strlen(symbol.filename());
+                    auto totalLen = nameLen + filenameLen + 1;
+                    auto newName = cast(char*) mem.xmalloc(totalLen);
+                    newName[totalLen] = 0;
+                    name[0] = '/';
+                    memcpy(newName, symbol.filename(), filenameLen);
+                    memcpy(newName[filenameLen], name, nameLen);
+                    mem.xfree(name);
+                    name = newName;
```

Comment:
If possible replace `if(blah blah) x = calculation()` with a function or nested function

Reply:
It performs a discrete task that should be tested properly. DMD is way too nested and way too branchy